### PR TITLE
Make local dev easier to get started on 

### PR DESCRIPTION
**Pull request by csakoda**
Addressing #184 

Adding a grab back of quality of life improvements for new devs (and educating myself along the way).  For context, I am new to NodeJS, if there are best practices I've not seen yet please just let me know.  I tried to hold back on factoring index.js (will take a swing in another PR perhaps), but that clearly would be valuable.  The changes I made to that file were to get the REPL experience going.

* Adding a `getting_started.md` README to give new devs an accelerated path
* Making googleapi auth optional (default: not set)
* Adding `get_live_data()` to seed the current JSON dataset into your REPL.
* Switch to using `module.exports` everywhere
    * I was having major trouble getting `exports` to work when doing `.load index.js` in the REPL.  If anyone has a best practice please LMK.
    * https://blog.tableflip.io/the-difference-between-module-exports-and-exports/

### CODE REVIEW

File: getting_started.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+```
+$ node
+> .load index.js
+> get_live_data()
+> dbl = toDataByLocation(static_data)
+```
+
+* (TODO) Running firebase locally
+
+## Architecture Overview
```
Comment (csakoda) [inline getting_started.md]: I bet I got parts of this wrong

File: getting_started.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+For now you can hack things into place.  After starting a NodeJS REPL, you can seed the current data and then use it in the REPL:
+
+```
+$ node
+> .load index.js
+> get_live_data()
+> dbl = toDataByLocation(static_data)
+```
+
+* (TODO) Running firebase locally
```
Comment (csakoda) [inline getting_started.md]: I see there's some setup already for the firebase emulator in our package.json for this

Comment (awong-dev) [reply getting_started.md]: it was autogenerated boilerplate.

File: getting_started.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+## Architecture Overview
+
+### Public Site
+
+* `index.html` reads `data.json`
+    * `data.json` is stored in a Firebase bucket, and is edited by running a firebase function that reads from Google Sheets and writes into Firebase.
```
Comment (awong-dev) [inline getting_started.md]: Authoritative data.json is stored in the bucket.  A copy is snapped into the hosting server so everything is just static file serving.

**Comment by awong-dev** [comment]
There's a few nits on the architecture section, but let's merge and fix in a followup. thank you for the cleanups!!!

**Pull request merged**